app/core/src/main/java/com/teamg/tourdeshot/core/mapper/ProductCategoryMapperImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.teamg.tourdeshot.core.mapper;
+
+
+import com.teamg.tourdeshot.core.api.productcategory.ProductCategoryDTO;
+import com.teamg.tourdeshot.core.mapper.utils.ProductCategoryMapperUtils;
+import com.teamg.tourdeshot.core.model.ProductCategory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductCategoryMapperImpl implements ProductCategoryMapper {
+
+    private final ProductCategoryMapperUtils utils;
+
+    @Autowired
+    public ProductCategoryMapperImpl(ProductCategoryMapperUtils utils) {
+        this.utils = utils;
+    }
+
+    @Override
+    public ProductCategoryDTO toCategoryDTO(ProductCategory productCategory) {
+        ProductCategoryDTO productCategoryDTO = new ProductCategoryDTO();
+        productCategoryDTO.setId(productCategory.getId());
+        productCategoryDTO.setName(productCategory.getName());
+        productCategoryDTO.setHasChild(utils.hasChild(productCategory));
+
+        if(productCategory.getParent() != null) {
+            productCategoryDTO.setParentId(productCategory.getParent().getId());
+        }
+        return productCategoryDTO;
```

Comment:
Here you can use a builder pattern. With if condition it can looks like 
`builder().setParentId(!Objects.isNull(productCategory.getParent()) ? productCategory.getParent().getId() : null`)`